src/test/java/io/strimzi/ValidatingWebhookTest.java

```diff
@@ -148,14 +163,18 @@ public void testEvictionOnPodWithExistingAnnotations() {
 
     @Test
     public void testWrongLabel() {
-        String podName = "my-cluster-kafka-1";
-        Map label = Collections.singletonMap("mylabel", "foo");
-        when(podResource.get()).thenReturn(mockedPod(podName, true, false));
+        String podName = "my-cluster-koo";
+        final Map<String, String> labels = Map.of(
+                "strimzi.io/kind", "Kafka",
+                "strimzi.io/name", "my-cluster-foo"
+        );
+
+        when(podResource.get()).thenReturn(mockedPod(podName, true, false, labels));
         ArgumentCaptor<Pod> podCaptor = ArgumentCaptor.forClass(Pod.class);
         when(podResource.patch(podCaptor.capture())).thenReturn(new Pod());
 
         ValidatingWebhook webhook = new ValidatingWebhook(client, true, true);
-        AdmissionReview reviewResponse = webhook.webhook(reviewRequest(podName, false, label));
+        AdmissionReview reviewResponse = webhook.webhook(reviewRequest(podName, false, labels));
```

Comment:
We are missing assertions about annotations in a bunch of tests, including this one.